synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -520,9 +521,15 @@ async def _download_url(self, url, user):
                 # FIXME: we should calculate a proper expiration based on the
                 # Cache-Control and Expire headers.  But for now, assume 1 hour.
                 expires = ONE_HOUR
-                etag = headers["ETag"][0] if "ETag" in headers else None
+                etag = (
+                    headers[b"ETag"][0].decode("ascii") if b"ETag" in headers else None
+                )
```

Comment:
this is nominally a bugfix (`"Etag"` would never be in the headers dict), but since we don't do anything with the resulting value other than stick it in the database, I don't think it's worth calling out separately.